### DISCUSSION TITLE
not enough memory!

### DISCUSSION THREAD

**Discussion by xabadabazinho** [marcel-licence/esp32_midi_sampler#96] [Q&A]
Hi, using Arduino IDE 2.1.1, win11, 
#define ES8388_CFG_I2C  2
#define ES8388_CFG_I2S  4
following error here (thanks in advance for any help):

ets Jun  8 2016 00:22:57

rst:0x1 (POWERON_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13964
load:0x40080400,len:3600
entry 0x400805f0

Loading data
Firmware started successfully
Connect to ES8388 codec... 0x00: 0x05
0x01: 0x40
AdcCh1!
MixChAMPL!
S8388 setup finished!
Reg 0x00 = 0x05
Reg 0x01 = 0x40
Reg 0x02 = 0x00
Reg 0x03 = 0x00
Reg 0x04 = 0x3c
Reg 0x05 = 0x00
Reg 0x06 = 0x00
Reg 0x07 = 0x7c
Reg 0x08 = 0x00
Reg 0x09 = 0x88
Reg 0x0a = 0x50
Reg 0x0b = 0x02
Reg 0x0c = 0x0c
Reg 0x0d = 0x02
Reg 0x0e = 0x30
Reg 0x0f = 0x20
Reg 0x10 = 0x00
Reg 0x11 = 0x00
Reg 0x12 = 0x16
Reg 0x13 = 0xb0
Reg 0x14 = 0x32
Reg 0x15 = 0x06
Reg 0x16 = 0x00
Reg 0x17 = 0x18
Reg 0x18 = 0x02
Reg 0x19 = 0x22
Reg 0x1a = 0x00
Reg 0x1b = 0x02
Reg 0x1c = 0x08
Reg 0x1d = 0x00
Reg 0x1e = 0x1f
Reg 0x1f = 0xf7
Reg 0x20 = 0xfd
Reg 0x21 = 0xff
Reg 0x22 = 0x1f
Reg 0x23 = 0xf7
Reg 0x24 = 0xfd
Reg 0x25 = 0xff
Reg 0x26 = 0x1b
Reg 0x27 = 0xb8
Reg 0x28 = 0x28
Reg 0x29 = 0x28
Reg 0x2a = 0xb8
Reg 0x2b = 0x80
Reg 0x2c = 0x00
Reg 0x2d = 0x00
Reg 0x2e = 0x1e
Reg 0x2f = 0x1e
Reg 0x30 = 0x1e
Reg 0x31 = 0x1e
Reg 0x32 = 0x00
Reg 0x33 = 0xaa
Reg 0x34 = 0xaa
I2S configured using following pins:
    BCLK,BCK: 5
    WCLK,LCK: 25
    DOUT: 26
    DIN: 35
MIDI listen on Serial with 115200 baud
Setup Serial2 with 31250 baud with rx: IO21 only
Setup MidiPort2 using Serial2
Total PSRAM: 4192059
Free PSRAM: 4015139
done!..
- Free PSRAM: 2988631
done!..
- Free PSRAM: 1962123
done!..
- Free PSRAM: 935615
not enough memory!
ESP.getFreeHeap() 257124
ESP.getMinFreeHeap() 257092
ESP.getHeapSize() 289648
ESP.getMaxAllocHeap() 110580
Total heap: 289648
Free heap: 257124
Total PSRAM: 4192011
Free PSRAM: 935615
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x400d3d2f  PS      : 0x00060c30  A0      : 0x800d3f9d  A1      : 0x3ffcafe0  
A2      : 0x3ffc404c  A3      : 0x3ffc3f8c  A4      : 0x3ffc3f14  A5      : 0x3ffc4340  
A6      : 0x3ffc4330  A7      : 0x3ffc4364  A8      : 0x00000000  A9      : 0x0000000c  
A10     : 0x3ffc3f20  A11     : 0x00000003  A12     : 0x00000000  A13     : 0x00000000  
A14     : 0x00000004  A15     : 0x000000ff  SAR     : 0x00000015  EXCCAUSE: 0x0000001d  
EXCVADDR: 0x00000000  LBEG    : 0x400d3d18  LEND    : 0x400d3ddc  LCOUNT  : 0x00000000
Core  1 register dump:
PC      : 0x400d3d2f  PS      : 0x00060c30  A0      : 0x800d3f9d  A1      : 0x3ffcafe0  
A2      : 0x3ffc404c  A3      : 0x3ffc3f8c  A4      : 0x3ffc3f14  A5      : 0x3ffc4340  
A6      : 0x3ffc4330  A7      : 0x3ffc4364  A8      : 0x00000000  A9      : 0x0000000c  
A10     : 0x3ffc3f20  A11     : 0x00000003  A12     : 0x00000000  A13     : 0x00000000  
A14     : 0x00000004  A15     : 0x000000ff  SAR     : 0x00000015  EXCCAUSE: 0x0000001d  
EXCVADDR: 0x00000000  LBEG    : 0x400d3d18  LEND    : 0x400d3ddc  LCOUNT  : 0x00000000  

Backtrace: 0x400d3d2c:0x3ffcafe0 0x400d3f9a:0x3ffcb010 0x400d3feb:0x3ffcb050 0x400d8cc5:0x3ffcb070

ELF file SHA256: f5dbcb6b587a04d4

Rebooting...

-------------------------------------------------------------------------------------------------------------------------
![image](https://github.com/marcel-licence/esp32_midi_sampler/assets/25510866/511b1f91-44b3-4c47-9cd4-0c423d1338eb)

**Comment by marcel-licence** [comment]
Hi, did you reduce the sample buffer size? It looks like that using the same PSRAM chip with different ESP32 arduino board versions results sometimes in less available memory.

**Comment by xabadabazinho** [comment]
I am truly sorry, was in the wrong discussion, went to the multitrack looper discussion and the info was there. Already solved, thank you, fastastic library!
